### PR TITLE
TP-939: Reduce log clogging by only logging progress every 30 sec

### PR DESCRIPTION
**Pull request by pativa**
* Should work when called in parallel with AtomicReference `compareAndSet`

### CODE REVIEW

**Review by askoog — APPROVED** [review]
minor comment on copyright header, other than that: 👍

File: ymer/src/main/java/com/avanza/ymer/LogWithInterval.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2015 Avanza Bank AB
```
[the remark this answers is not in the document]
Comment (pativa) [reply ymer/src/main/java/com/avanza/ymer/LogWithInterval.java:2]: The project validates that the header year equals 2015 😓  Maybe should fix that

[the remark this answers is not in the document]
Comment (pativa) [reply ymer/src/main/java/com/avanza/ymer/LogWithInterval.java:2]: I'm not sure if changing the year is correct? Anyway, the license plugin validates this year and it's the year we use in all other source files 😄

**Comment by pativa** [comment]
Most of the changes here are now from @ath0s suggested changes in https://github.com/AvanzaBank/ymer/pull/75 so maybe you can have a quick look at it again @askoog ?

File: ymer/src/main/java/com/avanza/ymer/RepeatingTask.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+
+class RepeatingTask implements AutoCloseable{
+	private final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
```
Comment (askoog) [inline ymer/src/main/java/com/avanza/ymer/RepeatingTask.java:26]: I suggest to use deamon threads with a descriptive name here

File: ymer/src/test/java/com/avanza/ymer/RepeatingTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		}
+
+		assertThat(counter.sum(), is(10L));
+	}
+
+	@Test
+	public void shouldNotTriggerAfterClosed() throws Exception {
+		LongAdder counter = new LongAdder();
+
+		try(RepeatingTask ignore = new RepeatingTask(Duration.ZERO, Duration.ofMillis(100), counter::increment)) {
```
Comment (askoog) [inline ymer/src/test/java/com/avanza/ymer/RepeatingTaskTest.java:44]: maybe use larger interval than 100ms to make test more stable? sleep(5) + sleep(10) with some thread context switching might sum up to +100ms if we are unlucky and have a build server under high load

**Review by askoog — APPROVED** [review]
Nice change! I think this is a better solution. Some minor comments below

File: ymer/src/test/java/com/avanza/ymer/RepeatingTaskTest.java (hunk 3 above)
Comment (ath0s) [reply ymer/src/test/java/com/avanza/ymer/RepeatingTaskTest.java]: I've loosened the timings a bit in https://github.com/AvanzaBank/ymer/pull/74/commits/b87f6421851adf2d546ad9ec31d91bfc3ea4dd0a

File: ymer/src/main/java/com/avanza/ymer/RepeatingTask.java (hunk 2 above)
Comment (ath0s) [reply ymer/src/main/java/com/avanza/ymer/RepeatingTask.java]: Fixed in https://github.com/AvanzaBank/ymer/pull/74/commits/b87f6421851adf2d546ad9ec31d91bfc3ea4dd0a

Comment (pativa) [reply ymer/src/main/java/com/avanza/ymer/RepeatingTask.java]: Looks good to me! 👍

**Comment by pativa** [comment]
Are we ready to merge this? 😃

**Review by askoog — APPROVED** [review]
ready to merge!

**Pull request merged**